concepts/class-inheritance/about.md

```diff
@@ -1,2 +1,291 @@
+           self.__displayDetails() 
+ 
+# creating object   
+obj = Student("ABC", 122333, "Physics")
+ 
+# accessing public member function of the class
+obj.accessPrivateFunction()
+```
+`__name`, `__roll`, and `__division` are private members of the class `Student`, while the `__displayDetails()` method is a private member function (accessible only within the class), and the `accessPrivateFunction()` method is a public member function of the class `Student` that can be accessed from anywhere within the programme. The `accessPrivateFunction()` method gives us access to the `Student` class's private members.
+### Class Functions
+The member functions of a class can be categorized into following two categories:
+
+__1. Accessor Functions__: These are those number functions that allow us to access data fields of the object. However, these functions cannot/do not change the value of the data members.
+
+```python
+class Student:
+    # private members
+     __name = None
+     __roll = None
+     __marks = None
+     __div = None
+     
+     # constructor
+    def __init__(self, name, roll, marks, div): 
+        self.__name = name
+        self.__roll = roll
+        self.__marks = marks
+        self.__div = div
+     
+     # public member function 
+     def displayDetails(self):
+         # accessing private data members
+         print("Name: ", self.__name)
+         print("Roll: ", self.__roll)
+         print("Marks:", self.__marks)
+         print("Division: ", self.__div)
+         
+     # Accessor Function    
+     def getRollNo(self):
+        return self.__roll
+     
+     # Mutator Function
+     def setDiv(self):
+        if self.__marks>=60:
+            self.__div = 'A'
+        elif self.__marks<60:
+            self.__div = 'B'
+        return self.__div
+```
+__2. Mutator Function__: This method is used to change the state of an object, i.e. it changes the data variable's hidden value. It has the ability to change the value of a variable on the fly. The update method is another name for this function. We can also use the word _set_ to name these procedures.
+
+As per the above example, `getRollNo()` is an accessor function and `setDiv` is a mutator function.
+
+## Inheritance
+
+## Single Inheritance
+
+When a subclass inherits only from one base class, it is known as single inheritance.
+
+```python
+#Base class
+class Odin:
+    def func1(self):
+        print("Odin: father of thor")
+
+#Derived/Subclass class
+class Thor(Odin):
+    def func2(self):
+        print("Thor: son of Odin")
+
+
+object = Thor()
+object.func1()
+object.func2()
+```
+The derived class will print both the statements of `func1` and `func2`.
+
+## Multiple Inheritance
+As we've seen, a class inherits from another class in this scenario. On the other side, multiple inheritance is a feature that allows a class to inherit characteristics and methods from many parent classes.
+```python
+class SubclassName(BaseClass1, BaseClass2, ...):
+    pass
+```

```

Comment:
I think this section needs to be extended to talk about how multiple inheritance works -- and how the parent classes are read in.  I think it might also be helpful to limit the initial example to only two base classes - one the "core" class, and one the "secondary", to help lead into the diamond problem and then to why mixins are a thing.

As with the single inheritance section above, I think it's important here to show how class attributes, instance attributes, and methods are inherited, and from there, how those get extended or overridden in the child class, so more code examples, and more narrative.